ili/utils/ndes_pt.py

```diff
@@ -262,41 +268,52 @@ def load_nde_lampe(
             Defaults to True.
         theta_normalize (bool, optional): whether to z-normalize theta.
             Defaults to True.
+        engine (str, optional): dummy argument to match sbi interface.
+            Must be set to 'NPE' or will be overwritten.
         **model_args: additional arguments to pass to the model.
     """
+    if 'NPE' not in engine:
+        logging.warning(f'Engine {engine} not supported in lampe backend. '
+                        'Continuing as if to engine=NPE.')
+
     if model == 'mdn':  # for mixture density networks
         if not (set(model_args.keys()) <= {'hidden_features', 'num_components'}):
             raise ValueError(f"Model {model} arguments mispecified.")
+        model_defaults = dict(hidden_features=16, num_components=3)
+        model_args = {**model_defaults, **model_args}
         model_args['hidden_features'] = [model_args['hidden_features']] * 3
         model_args['components'] = model_args.pop('num_components', 2)
         flow_class = zuko.flows.mixture.GMM
-    elif model == 'cnf':  # for continuous flow models
-        # number of time embeddings
-        model_args['hidden_features'] = [
-            model_args['hidden_features']] * 2
-        model_args['freqs'] = model_args.pop('num_transforms', 2)
-        flow_class = zuko.flows.continuous.CNF
-    else:  # for all discrete flow models
-        if not (set(model_args.keys()) <= {'hidden_features', 'num_transforms'}):
-            raise ValueError(f"Model {model} arguments mispecified.")
-        model_args['hidden_features'] = [
-            model_args['hidden_features']] * 2
-        model_args['transforms'] = model_args.pop('num_transforms', 2)
-
-        if model == 'maf':
-            flow_class = zuko.flows.autoregressive.MAF
-        elif model == 'nsf':
-            flow_class = zuko.flows.spline.NSF
-        elif model == 'nice':
-            flow_class = zuko.flows.coupling.NICE
-        elif model == 'gf':
-            flow_class = zuko.flows.gaussianization.GF
-        elif model == 'sospf':
-            flow_class = zuko.flows.polynomial.SOSPF
-        elif model == 'naf':
-            flow_class = zuko.flows.neural.NAF
-        elif model == 'unaf':
-            flow_class = zuko.flows.neural.UNAF
+    else:
+        model_defaults = dict(hidden_features=16, num_transforms=2)
+        model_args = {**model_defaults, **model_args}
+        if model == 'cnf':  # for continuous flow models
+            # number of time embeddings
+            model_args['hidden_features'] = [
+                model_args['hidden_features']] * 2
+            model_args['freqs'] = model_args.pop('num_transforms', 2)
+            flow_class = zuko.flows.continuous.CNF
+        else:  # for all discrete flow models
+            if not (set(model_args.keys()) <= {'hidden_features', 'num_transforms'}):
+                raise ValueError(f"Model {model} arguments mispecified.")
```

Comment:
This can never happen given we have now added defaults for these arguments (and thus this line is never used in the unit tests). So please delete this check

Reply:
You're right, we should've done this check before the default setting. I've updated this

Reply:
I've also done the same ordering for the pydelfi loader